binary-codec/types/st_object.go

```diff
@@ -1,28 +1,55 @@
 package types
 
 import (
-	"fmt"
+	"reflect"
 	"sort"
 
+	"github.com/mitchellh/mapstructure"
 	"github.com/xyield/xrpl-go/binary-codec/definitions"
 	"github.com/xyield/xrpl-go/binary-codec/serdes"
 )
 
+// FieldMutation allows values to mutated before being serialized.
+type FieldMutation func(any) any
+
+// Zero returns a FieldMutation that sets the value to its zero value.
+func Zero() FieldMutation {
+	return func(v any) any {
+		return reflect.Zero(reflect.TypeOf(v)).Interface()
+	}
+}
+
 // STObject represents a map of serialized field instances, where each key is a field name
 // and the associated value is the field's value. This structure allows us to represent nested
 // and complex structures of the Ripple protocol.
-type STObject struct{}
+type STObject struct {
+	OnlySigning bool
+	Mutations   map[string]FieldMutation
+}
 
 // FromJson converts a JSON object into a serialized byte slice.
 // It works by converting the JSON object into a map of field instances (which include the field definition
 // and value), and then serializing each field instance.
 // This method returns an error if the JSON input is not a valid object.
 func (t *STObject) FromJson(json any) ([]byte, error) {
```

Comment:
I did have issues with certain fields, and changed this to use json encoding rather than mapstructure
Needs to add support for ledger object/tx type string as well
You can see my commit here for reference https://github.com/CreatureDev/xrpl-go/commit/842ff1f97d444e2da4530cdaa28429db6635a83b

Reply:
relevant PR https://github.com/CreatureDev/xrpl-go/pull/4